General update of technical docs (#196)
digitalbitbox/bitbox-base: docs/applications/bitcoin-core.md

### Configuration

> The application configuration is specified in the local `/etc/bitcoin/bitcoin.conf` file. The `bitcoin.conf` file is created from a template into the Armbian image during the build process, and can be automatically updated during regular operations as well. These are the initial settings:

Reviewer: ```suggestion
The application configuration is specified in the local `/etc/bitcoin/bitcoin.conf` file. The `bitcoin.conf` file is generated from a template during the build process of the Armbian image. It can updated during regular operations. These are the initial settings:
```

seednode=nkf5e6b7pl4jfd4a.onion
seednode=xqzfakpeuvrobvpj.onion
seednode=tsyvzsqwa2kkf6b2.onion

# validation
reindex-chainstate=0
```

Some notes about this specific configuration:

* **Network options**
  * `mainnet`/`testnet`: the build script defaults to building a mainnet node, but can be reconfigured by:
    * specifying the corresponding build parameter in [`build.conf`](../../armbian/base/build/build.conf)

>     * or running the command `bbb-config.sh set bitcoin_network mainnet` manually on the BitBox Base (see [Operating System/Helper Scripts](../os/helper-scripts.md)).

Reviewer: Should that be `testnet` here?
```suggestion
    * or running the command `bbb-config.sh set bitcoin_network testnet` manually on the BitBox Base (see [Operating System/Helper Scripts](../os/helper-scripts.md)).
```

* **Server options**
  * `server`: enables the RPC interface
  * `listen`: accept connections from other nodes
  * `listenonion`: create a Tor hidden service and accept incoming connections from other nodes on that address
  * `daemon`: to caputre all log output with the `printtoconsole` option, bitcoind must not run as a daemon
  * `txindex`: a full transaction index is not necessary, as we use electrs with its own indices to serve transaction information
  * `prune`: both c-lightning as well as electrs do not fully support pruned Bitcoin nodes at the moment
  * `disablewallet`: wallet functionality of bitcoind is not used and therefore disabled

* **RPC options**
  * `rpconnect`: specifying `127.0.0.1`, the bitcoind api listens only for local connections 
  * `rpcport`: the bitcoind api always listens on port `8334`, both for mainnet and testnet 

>   * `rpcauth`: authentication to bitcoind api uses the `rpcauth` method, with clients using static `rpcuser` and `rpcpassword` values. Credentials are created using an [adapted version](TODO) of the Bitcoin Core [rpcauth.py](TODO) Python script

Reviewer: Here are two TODOs that might have been forgotten, if not just ignore.

Author: Thanks, fixed all three TODOs

* **Performance options**
  * `dbcache`: initially, the database cache is optimized for the initial block download (IBD), allocating 2GB memory. This value is set to a lower value (default: 300 MB) after bitcoind is fully synced.
  * `maxconnections`: upper limit number of connections to other Bitcoin nodes
  * `maxuploadtarget`: upper limit for daily upload data volume

* **Tor**
  * `proxy`: set bitcoind to also use the Tor proxy for IPv4 and IPv6 connections
  * `seednode`: initial Tor nodes to bootstrap connections and discover additional nodes

* **Validation**

>   * `reindex-chainstate`: if this option is set to `1`, e.g. by the script [`bbb-config.sh`](TODO), bitcoind discards the current UTXO set on restart and revalidates the whole blockchain from Genesis.

Reviewer: TODO ?
